### DISCUSSION TITLE
Is it possible to track metrics without using DVC Pipelines?

### DISCUSSION THREAD

**Discussion by AlexandreBrown** [treeverse/dvc#7371] [Help] [answered]
Hello, we already use another pipeline orchestrating tool (Kubeflow) but are looking into using DVC to version our datasets and maybe metrics.  
  
Currently we already log metrics using Kubeflow, it generates a json with the metrics (in a key value pairs format style) and gets uploaded to a metadata db.  
Long story short, I know we could use DVC to keep track of the metadta json file but I'm wondering if we could also use DVC metrics feature to see diff etc if we use metrics not generated by DVC?   

Is there a specific format the json needs to be in in order to work with DVC metrics or does dvc diff simply reads it as key value pairs (eg: display key, display value etc) ?
  
Thanks

**Comment by daavoo** [comment, marked as answer]
You should be able to use [dvc metrics](https://dvc.org/doc/command-reference/metrics) commands using as `targets` any file in any of the [supported formats ](https://dvc.org/doc/command-reference/metrics#supported-file-formats) even if those files are not associated to a DVC pipeline:

```
git init
dvc init
echo '{"foo": 1}' > foo.json
git add .
git commit -m "1"
dvc metrics show foo.json
```
```
Path      foo                                                         
foo.json  1
```

```
echo '{"foo": 2}' > foo.json
git add foo.json
git commit -m "2"
dvc metrics diff --targets foo.json
```

```
Path      Metric    HEAD    workspace    Change                       
foo.json  foo       1       2            1
```
